### DISCUSSION TITLE
[Question]How do I write test code using pytest?

### DISCUSSION THREAD

**Discussion by NaotoTakata** [flet-dev/flet#1944] [Q&A]
### Question

Hi flet community

I would like to ask how to write test code for a Flet application using pytest. 
I did a quick search, but there's not a lot of information about it.

Let's say my application is as follows:

I want to make sure that `self.message.value` is "Water" when you run the app, and after clicking `self.btn`, `self.message.value` should be "Coffee."

### Code sample

```python
import flet as ft


class SubClass(ft.UserControl):
    def __init__(self, page: ft.Page):
        super().__init__()
        self.page = page
        self.message = ft.Text("Water")

        self.btn = ft.ElevatedButton(text="change to Coffee", on_click=self.clicked_btn)

    def clicked_btn(self, e):
        if self.message.value == "Coffee":
            self.message.value = "Water"
            self.btn.text = "change to Coffee"
        else:
            self.message.value = "Coffee"
            self.btn.text = "change to Water"

        self.message.update()
        self.btn.update()

    def build(self):
        return ft.Column(controls=[self.message, self.btn])


def main(page: ft.Page):
    subclass = SubClass(page)
    page.add(subclass)
    page.update()


if __name__ == "__main__":
    ft.app(target=main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by dstalzjohn** [comment]
I am confronted with the same problem and didn’t find a solution. What doesn’t work is the use of playwright or something similar because they work with frameworks based on node.js.

**Comment by DanMarqz** [comment]
Hello,

I don't know if pytest can be used to perform tests in Flet, however, in the [CONTRIBUTING.md](https://github.com/flet-dev/flet/blob/47fc5f93d567d82e28cab07bbd24614ae36865de/CONTRIBUTING.md#running-tests) file there is a section that talks about how to perform tests on the fork created to contribute, I leave the document here so you can view it.

Greetings.

**Reply by dstalzjohn** [reply]
Thank you for your comment. Unfortunately, I only found tests written in dart. What we need is testability of our application written in flet using pytests. 

The only possibility I currently see here are screenshot tests which would mean a lot of overhead.
